main/internal/implementations/terraform_workspace/search_repo_workspaces_test.go

```diff
@@ -0,0 +1,128 @@
+package terraformWorkspace
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTerraformCloud_cleanDirectoryName(t *testing.T) {
+	type args struct {
+		directory string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "happy path",
+			args: args{
+				directory: "google-backend-api-dev",
+			},
+			want: "google-backend-api-dev",
+		},
+		{
+			name: "happy path with trailing slash",
+			args: args{
+				directory: "google-backend-api-dev/",
+			},
+			want: "google-backend-api-dev",
+		},
+		{
+			name: "happy path with trailing slash and whitespace",
+			args: args{
+				directory: "google-backend-api-dev/ ",
+			},
+			want: "google-backend-api-dev",
+		},
+		{
+			name: "happy path with trailing at beginning and end",
+			args: args{
+				directory: "/google-backend-api-dev/",
+			},
+			want: "google-backend-api-dev",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert.Equalf(t, tt.want, cleanDirectoryName(tt.args.directory), "cleanDirectoryName(%v)", tt.args.directory)
+		})
+	}
+}
+
+func TestExtractBackendDetails(t *testing.T) {
+	// Given
+	ctx := context.Background()
+
+	filePath := "test.tfstate"
+	fileContent := []byte(`
+			terraform {
+			  backend "gcs" {
+				bucket  = "my-gcs-terraform-backend-bucket"
+				prefix  = "terraform/state"
+			  }
+			}
+		`)
+	err := os.WriteFile(filePath, fileContent, 0600)
+	require.NoError(t, err)
```

Comment:
We should not need to actually write the file in order to run Decode: https://pkg.go.dev/github.com/hashicorp/hcl/v2/hclsimple#Decode, just give it a dummy filepath name